### ISSUE TITLE
How to index datetime value with JSON?

### ISSUE THREAD

**Issue by mosuka** [quickwit-oss/tantivy#719]
I want to index datetime value with JSON.
In the schema, the following date field has created with the name `timestamp`:

```
...
  {
    "name": "timestamp",
    "type": "date",
    "options": {
      "indexed": true,
      "stored": true,
      "fast": "single"
    }
  }
...
```

When I index JSON with the timestamp value set as follows:

```
"timestamp" : "2019-12-06T12:22:54.341Z",
```

The following error occurred:

```
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: ValueError("timestamp", TypeError("Expected an integer, got String(\"2019-12-06T12:22:54.341Z\")"))', src/libcore/result.rs:1084:5
stack backtrace:
   0: std::panicking::default_hook::{{closure}}
   1: std::panicking::default_hook
   2: std::panicking::rust_panic_with_hook
   3: std::panicking::continue_panic_fmt
   4: rust_begin_unwind
   5: core::panicking::panic_fmt
   6: core::result::unwrap_failed
   7: bayard::server::server::IndexServer::apply_entry
   8: bayard::server::server::IndexServer::async_applier::{{closure}}
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

Since integer data is expected, change the timestamp value to an integer value as follows:

```
"timestamp" : 1575647612000,
```

The following error occurred:

```
thread 'thrd-tantivy-index0' panicked at 'This is not a date field.', /Users/m-osuka/.cargo/registry/src/github.com-1ecc6299db9ec823/tantivy-0.10.3/src/schema/value.rs:167:18
stack backtrace:
   0: std::panicking::default_hook::{{closure}}
   1: std::panicking::default_hook
   2: std::panicking::rust_panic_with_hook
   3: std::panicking::begin_panic
   4: tantivy::indexer::segment_writer::SegmentWriter::add_document
   5: tantivy::indexer::index_writer::index_documents
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

What format is good for expressing datetime in JSON?

Tantivy version is 0.10.3.

**Comment by fulmicoton** [comment]
This is a bug. Let's fix it for 0.11 which should be released soonish.

**Comment by mosuka** [comment]
Thank you for your reply!
